llms/mlx_lm/models/llama.py

```diff
@@ -45,7 +46,8 @@ def __init__(self, args: ModelArgs):
         self.n_heads = n_heads = args.num_attention_heads
         self.n_kv_heads = n_kv_heads = args.num_key_value_heads
 
-        head_dim = args.hidden_size // n_heads
+        self.head_dim = head_dim = args.head_dim if args.head_dim else args.hidden_size // n_heads
```

Comment:
Very minor nit, this is a nice idiom you can use in Python for when something is `None`

```suggestion
        self.head_dim = head_dim = args.head_dim or args.hidden_size // n_heads
```

Reply:
Cool, I didn't know that.